HIP/hip-906.md

```diff
@@ -0,0 +1,155 @@
+
+The following table describes the function selector for the new `hbarAllownace` and `hbarApprove` functions and the associated function signature and response.
+
+| Function Selector Hash | Function Signature                           | Response                                                                               | 
+|------------------------|----------------------------------------------|----------------------------------------------------------------------------------------|
+| 0xbbee989e             | hbarAllowance(address spender)               | (ResponseCode, uint256 - amount of hbar allowances currently available to the spender) | 
+| 0x76f17392             | hbarApprove(spender address, amount uint256) | ResponseCode                                                                           |
+
+The solidity interface for IHRC632 will be the following
+
+```
+interface IHRC632 {
+    function hbarAllowance(address spender) external returns (responseCode, int256);
+    function hbarApprove(spender address, amount int256) external returns (responseCode);
+}
+```
+
+Once the above functionality has been implemented in services, an EOA or contract with hbars will be able to call the `hbarAllownace` and `hbarApprove` functions as follows:
+
+```
+IHRC(accountAddress).hbarAllowance(address spender)
+IHRC(accountAddress).hbarApprove(spender address, amount int256)
+```
+> The `hbarApprove` function will be callable by either an EOA or a contract as long as the signatures required by the security model are met.
+
+## Key Benefits and Possible Use Cases
+Key benefits include:
+1. Better developer experience.  Develops will not have to use the SDK to perform tasks that will be available via this HIP.
+2. Support for potentially new flows for DAPs such as DEFI and token marketplaces.
+3. Differentiates Hedera from other chains that do not allow an allowance/approval mechanism for their native tokens.
+
+Potential use cases that have been communicated by the community include uses for NFT marketplaces to overcome signing requirements that can cause friction.
+For example in a marketplace application, a contract can act as an agent for an EOA to hold balances of hbar, fungible tokens and non-fungible tokens.  This HIP enables this contract to
+grant an allowance to another contract who can act as a broker by gathering all the necessary allowances to enable an exchange.  Using the allowance 
+mechanism can alleviate the need for an EOA to sign transactions for such use cases as royalty fallback fees.
+
+## Updates via HAPI
+
+It is possible that the approved allowances for a spender can be spent or increased by submitting a separate approval transaction via
+HAPI.  In this case, the smart contract would not be aware of these changes in state could possibly hold stale state.  While it would be possible
```

Comment:
This sentence is not fully clear. Probably you meant:

`would not be aware of these changes in state and it could possibly hold stale state`?